### PR TITLE
Add support for publisher confirms for producers

### PR DESCRIPTION
**Pull request by bmichalski**
This PR attempts to solve #460 (and #74 even though it's been closed for no apparent reason by reporter).
It allows to turn confirm mode on for a producer's channel.

### CODE REVIEW

**Comment by UgoTurner** [comment]
Hello, it's been a while since that merge request was published. Does someone know why it is on standby ?

**Comment by bmichalski** [comment]
If any change is needed for this PR to be merged, I'd be happy to comply :)
Also it seems to me that this is a pretty important feature: without it there is no way to quickly make sure that the message has been received and persisted by the broker (in order to survive a restart for example).

**Comment by erild** [comment]
Hello, i'm interested in this feature as well.
I manage to test your branch without issue so there doesn't seem to be an issue with the code itself.

Is there any way to know if the maintainers are planning to include it soon ?

**Comment by umulmrum** [comment]
+1 :-)

**Comment by redshark1802** [comment]
would love to have this feature, is there any reason this pr is still open?

**Comment by redshark1802** [comment]
@skafandri @stloyd could you please look at this pr, it's a feature that several users want.

**Comment by ypereirareis** [comment]
Hi, waiting for this feature too !

File: RabbitMq/PublisherConfirmsProducer.php [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+{
+    protected $publisherConfirmsEnabled = false;
+    protected $publisherConfirmsTimeout = 3;
+
+    public function getPublisherConfirmsTimeout()
+    {
+        return $this->publisherConfirmsTimeout;
+    }
+
+    public function setPublisherConfirmsTimeout($publisherConfirmsTimeout)
```
Comment (prgTW) [inline RabbitMq/PublisherConfirmsProducer.php]: @bmichalski this method could be defined in an interface of some kind. When publisher is of custom class yet still `publisher_confirms: true` is used container extension adds a method call without checking if the method exists. This way we could make the code better if this method is introduced for a purpose (by implementing an interface). WDYT?

**Pull request closed without merging**